### PR TITLE
Database and user for db_clusters

### PR DESCRIPTION
**Pull request by schirevko**

### CODE REVIEW

File: examples/create-mcs-db-user/main.tf [hunk 1]
```diff
@@ -39,7 +39,7 @@ resource "mcs_db_instance" "db-instance" {
 
 resource "mcs_db_database" "db-database" {
   name        = "testdb"
-  instance_id = mcs_db_instance.db-instance.id
+  dbms_id = mcs_db_instance.db-instance.id
```
Comment (ftersin) [inline examples/create-mcs-db-user/main.tf:42]: add spaces

File: mcs/data_source_mcs_db_user.go [hunk 2]
```diff
@@ -22,6 +22,12 @@ func dataSourceDatabaseUser() *schema.Resource {
 			},
 
 			"instance_id": {
+				Type:       schema.TypeString,
+				Optional:   true,
+				Deprecated: "Please, use dmbs_id attribute instead",
+			},
```
Comment (ftersin) [inline mcs/data_source_mcs_db_user.go:28]: shouldn't we use here ConflictsWith as well?

Comment (ftersin) [reply mcs/data_source_mcs_db_user.go:28]: and why computed dbms_type is absent?

File: mcs/resource_mcs_db_database.go [hunk 3]
```diff
@@ -55,17 +70,37 @@ func resourceDatabaseDatabaseCreate(d *schema.ResourceData, meta interface{}) er
 	}
 
 	databaseName := d.Get("name").(string)
-	instanceID := d.Get("instance_id").(string)
+	dbmsIDRaw, dbmsIDOk := d.GetOk("dbms_id")
+	instanceIDRaw, instanceIDOk := d.GetOk("instance_id")
+	if !dbmsIDOk && !instanceIDOk {
+		return fmt.Errorf("one of instance_id or dbms_id must be set")
```
Comment (ftersin) [inline mcs/resource_mcs_db_database.go:76]: lets mention dbms_id only, as instance_id is deprecated

File: mcs/resource_mcs_db_database.go [hunk 4]
```diff
@@ -55,17 +70,37 @@ func resourceDatabaseDatabaseCreate(d *schema.ResourceData, meta interface{}) er
...
+		dbmsID = instanceIDRaw.(string)
+	} else {
+		dbmsID = dbmsIDRaw.(string)
+	}
 
-	instance, err := instanceGet(DatabaseV1Client, instanceID).extract()
+	dbmsResp, err := getDBMSResource(DatabaseV1Client, dbmsID)
 	if err != nil {
-		return fmt.Errorf("error while getting mcs_db_instance: %s", err)
+		return fmt.Errorf("error while getting resource: %s", err)
```
Comment (ftersin) [inline mcs/resource_mcs_db_database.go:87]: resource -> instance or cluster

File: mcs/resource_mcs_db_database.go [hunk 5]
```diff
@@ -128,10 +172,25 @@ func resourceDatabaseDatabaseRead(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	d.Set("name", databaseName)
+	if _, ok := d.GetOk("instance_id"); ok {
+		d.Set("instance_id", dbmsID)
+	}
+	if _, ok := d.GetOk("dbms_id"); ok {
+		d.Set("dbms_id", dbmsID)
```
Comment (ftersin) [inline mcs/resource_mcs_db_database.go:179]: should not we set this anyway?

File: mcs/resource_mcs_db_database.go [hunk 6]
```diff
@@ -128,10 +172,25 @@ func resourceDatabaseDatabaseRead(d *schema.ResourceData, meta interface{}) erro
...
+	}
 
 	return nil
 }
 
+func resourceDatabaseDatabaseUpdate(d *schema.ResourceData, meta interface{}) error {
+	_, dbmsIDOk := d.GetOk("dbms_id")
+	_, instanceIDOk := d.GetOk("instance_id")
+	if !dbmsIDOk && !instanceIDOk {
+		return fmt.Errorf("one of instance_id or dbms_id must be set")
```
Comment (ftersin) [inline mcs/resource_mcs_db_database.go:189]: mention dbms_id only

File: mcs/resource_mcs_db_user.go [hunk 7]
```diff
@@ -68,17 +82,37 @@ func resourceDatabaseUserCreate(d *schema.ResourceData, meta interface{}) error
 
 	userName := d.Get("name").(string)
 	rawDatabases := d.Get("databases").([]interface{})
-	instanceID := d.Get("instance_id").(string)
+	dbmsIDRaw, dbmsIDOk := d.GetOk("dbms_id")
+	instanceIDRaw, instanceIDOk := d.GetOk("instance_id")
+	if !dbmsIDOk && !instanceIDOk {
+		return fmt.Errorf("one of instance_id or dbms_id must be set")
```
Comment (ftersin) [inline mcs/resource_mcs_db_user.go:88]: mention dbms_id only

File: mcs/resource_mcs_db_user.go [hunk 8]
```diff
@@ -151,6 +193,13 @@ func resourceDatabaseUserRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("unable to set databases: %s", err)
 	}
 
+	if _, ok := d.GetOk("instance_id"); ok {
+		d.Set("instance_id", dbmsID)
```
Comment (ftersin) [inline mcs/resource_mcs_db_user.go:197]: do we need this at all after it's deprecated?

File: mcs/resource_mcs_db_user.go [hunk 9]
```diff
@@ -166,14 +215,21 @@ func resourceDatabaseUserUpdate(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("invalid mcs_db_user ID: %s", d.Id())
 	}
 
-	instanceID := userID[0]
+	_, dbmsIDOk := d.GetOk("dbms_id")
+	_, instanceIDOk := d.GetOk("instance_id")
+	if !dbmsIDOk && !instanceIDOk {
+		return fmt.Errorf("one of instance_id or dbms_id must be set")
```
Comment (ftersin) [inline mcs/resource_mcs_db_user.go:221]: dmbs_id only

File: examples/create-mcs-db-user/main.tf (hunk 1 above)
Comment (schirevko) [reply examples/create-mcs-db-user/main.tf]: fixed

File: mcs/resource_mcs_db_database.go (hunk 3 above)
Comment (schirevko) [reply mcs/resource_mcs_db_database.go]: fixed

File: mcs/resource_mcs_db_database.go (hunk 4 above)
Comment (schirevko) [reply mcs/resource_mcs_db_database.go]: fixed

File: mcs/resource_mcs_db_database.go (hunk 6 above)
Comment (schirevko) [reply mcs/resource_mcs_db_database.go]: fixed

File: mcs/resource_mcs_db_user.go (hunk 7 above)
Comment (schirevko) [reply mcs/resource_mcs_db_user.go]: fixed

File: mcs/resource_mcs_db_user.go (hunk 9 above)
Comment (schirevko) [reply mcs/resource_mcs_db_user.go]: fixed

**Review by ftersin — APPROVED** [review]

**Review by pashapdev — APPROVED** [review]

**Review by irlndts — APPROVED** [review]

**Pull request merged**